src/gazebo_mavlink_interface.cpp

```diff
@@ -587,6 +587,13 @@ void GazeboMavlinkInterface::OnUpdate(const common::UpdateInfo&  /*_info*/) {
     mavlink_interface_->pollForMAVLinkMessages();
   }
 
+  // We need to send out heartbeats at a high rate until the connection is established,
+  // otherwise PX4 on USB doesn't enable mavlink and the buffer fills up.
+  if ((current_time - last_heartbeat_sent_time_).Double() > 1.0 || !mavlink_interface_->ReceivedHeartbeats()) {
+    mavlink_interface_->SendHeartbeat();
+    last_heartbeat_sent_time_ = current_time;
+  }
```

Comment:
Do we still want to progress with the simulation when in this phase? or should we just terminate here?

Reply:
Good question. We could terminate but we would have to test against SITL with lockstep to make sure we don't break it.